packages/components/src/ConfirmationModal/ConfirmationModal.mdx

```diff
@@ -0,0 +1,102 @@
+---
+name: ConfirmationModal
+menu: Components
+route: /components/confirmation-modal
+---
+
+import { useRef, useState } from "react";
+import { Playground, Props } from "docz";
+import { ComponentStatus } from "@jobber/docx";
+import { Button } from "@jobber/components/Button";
+import { ConfirmationModal } from ".";
+
+# Confirmation Modal
+
+<ComponentStatus stage="pre" responsive="no" accessible="no" />
+
+A ConfirmationModal allows users to confirm or cancel actions that they are
+preforming. Examples of actions that may need confirmation are navigating off an
+edited page or deleting an object.
+
+```ts
+import {
+  ConfirmationModal,
+  confirmationModalReducer
+} from "@jobber/components/ConfirmationModal";
+```
+
+<Playground>
+  {() => {
+    const [open, setOpen] = useState(false);
+    return (
+      <>
+        <Button label="Open" onClick={() => setOpen(true)} />
+        <ConfirmationModal
+          title="Should we?"
+          text="Do something…"
+          confirmLabel="Do it"
+          open={open}
+          onConfirm={() => alert("✅")}
+          onCancel={() => alert("🙅‍♂️")}
+          onRequestClose={() => setOpen(false)}
+        />
+      </>
+    );
+  }}
+</Playground>
+
+## Design & Usage Guidelines
+
+ConfirmationModals should be used to confirm or cancel an action the user is
+preforming.
+
+- Ensure that you are labeling the buttons with appropriate verbs. For example:
+  "Are you sure you want to delete" should have a label of "Delete" which is the
+  action that will be preformed.
+- It's good to identify the Object that is being acted upon. For example: "Are
+  you sure you want to delete the user: Bub"
+
+## Props
+
+<Props of={ConfirmationModal} />
+
+## Controlled Confirmation Modal
+
+In some instances, such as a list, you may want to place a single Confirmation
```

Comment:
I think we need a little more instruction here about what the code is below. I.e., how this solution can be used as a template for other solutions.
Is there a way we can make this a component or a function of its own?

Reply:
Added some more description here.